auth.php

```diff
@@ -603,14 +605,14 @@ protected function get_user_fields_parameters() {
 
         $mappingfield = $this->get_mapping_field();
         if ($this->should_create_user() || $this->should_update_user()) {
-            $parameters['firstname'] = new external_value(PARAM_NOTAGS, 'The first name(s) of the user', VALUE_OPTIONAL);
-            $parameters['lastname']  = new external_value(PARAM_NOTAGS, 'The family name of the user', VALUE_OPTIONAL);
+            $parameters['firstname'] = new external_value(core_user::get_property_type('firstname'), 'The first name(s) of the user', VALUE_OPTIONAL);
```

Comment:
Looks like this function is for 3.1+

Reply:
This shouldn't be in there - I fudged the merge from the Moodle code base into this plugin repo. I've taken these changes out.